plugins/module_utils/common.py

```diff
@@ -173,6 +172,9 @@ def auth_set(*names):
         if key in AUTH_ARG_MAP.keys() and value is not None:
             if key == 'api_key':
                 setattr(configuration, key, {'authorization': "Bearer {0}".format(value)})
+            elif key == 'proxy_headers':
+                headers = urllib3.make_headers(proxy_basic_auth=value)
```

Comment:
Do you think we can have other `proxy_headers` like `user-agent`?

Reply:
@Akasurde could you please clarify ?

Reply:
I think `proxy_headers` can be a `dict` instead of `str` like

```
    proxy_headers:
       proxy_basic_auth: 'username:password'
       user_agent: 'Batman'
```

Something similar to 
https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html?highlight=proxy_headers#urllib3.util.make_headers